src/transformers/optimization.py

```diff
@@ -165,6 +165,53 @@ def lr_lambda(current_step):
     return LambdaLR(optimizer, lr_lambda, last_epoch)
 
 
+def get_polynomial_decay_schedule_with_warmup(
+    optimizer, num_warmup_steps, num_training_steps, lr_end=1e-7, power=2.0, last_epoch=-1
+):
+    """
+    Create a schedule with a learning rate that decreases as a polynomial decay
+    from the initial lr set in the optimizer to end lr defined by `lr_end`,
+    after a warmup period during which it increases linearly from 0 to the
+    initial lr set in the optimizer.
+
+    Args:
+        optimizer (:class:`~torch.optim.Optimizer`):
+            The optimizer for which to schedule the learning rate.
+        num_warmup_steps (:obj:`int`):
+            The number of steps for the warmup phase.
+        num_training_steps (:obj:`int`):
+            The total number of training steps.
+        lr_end (:obj:`float`, `optional`, defaults to 1e-08):
+            The end LR.
+        power (:obj:`float`, `optional`, defaults to 1.0):
```

Comment:
```suggestion
        power (:obj:`float`, `optional`, defaults to 2.0):
```
Same.

Reply:
thank you for catching those inconsistencies, @sgugger .

fairseq implementation's default is actually power=1.0 and not 2.0, whereas others use 2.0. 
https://github.com/pytorch/fairseq/blob/master/fairseq/optim/lr_scheduler/polynomial_decay_schedule.py#L37

power=1.0 kind of defeats the purpose behind polynomial. 

Which do you recommend we use as the default? Especially since we currently don't have the cl arg to override the default if it's used for example scripts.

Reply:
Agreed. and also please see https://github.com/huggingface/transformers/pull/6361#issuecomment-672287450.

may be we can merge at 2.0 and adjust later if need be. this is a default after all.

Reply:
Agreed.